src/js/components/modals/JobFormModal.js

```diff
@@ -150,29 +155,92 @@ class JobFormModal extends mixin(StoreMixin) {
   }
 
   handleJSONEditorChange(jsonDefinition) {
-    try {
-      let job = new Job(JSON.parse(jsonDefinition));
-      this.setState({errorMessage: null, job});
-    } catch (error) {
-      // TODO: DCOS-7734 Handle error
-    }
+
+    // Keep the new JSON string in the JSON definition
+    // Validation will occur upon submission of the form
+    this.setState({
```

Comment:
Why do we need to change this method? What's wrong with converting the JSON definition to an `Job` struct?